geode-core/src/main/java/org/apache/geode/distributed/internal/InternalDistributedSystem.java

```diff
@@ -172,6 +167,16 @@
 
   private final StatisticsManager statisticsManager;
   private final FunctionStatsManager functionStatsManager;
+  /**
+   * True if the user is allowed lock when memory resources appear to be overcommitted.
+   */
+  private final boolean allowMemoryLockWhenOvercommitted =
+      Boolean.getBoolean(GEMFIRE_PREFIX + "Cache.ALLOW_MEMORY_OVERCOMMIT");
+  /**
+   * True if memory lock is avoided when memory resources appear to be overcommitted.
+   */
+  private final boolean avoidMemoryLockWhenOvercommitted =
```

Comment:
Why would a user ever set both to TRUE?  It seems like one is saying lock memory when over committed and the other says don't lock memory when over committed...

Reply:
Yes, user should not set both system properties in a cache. These two system properties mean different things. User should only set one of them. However, if the user sets both, Geode will not lock memory to avoid system experience unexpected consequences by locking memory without enough memory.